app/models/task.py

```diff
@@ -2,4 +2,39 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, default=None)
+    goal = db.relationship("Goal", back_populates="tasks")
+    goal_id_parent = db.Column(db.Integer, db.ForeignKey("goal.goal_id"))
```

Comment:
This works! Consider `goal_id` and remove the `parent` in order to stick to common naming conventions.